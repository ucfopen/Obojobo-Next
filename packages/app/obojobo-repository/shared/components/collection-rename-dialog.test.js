import React from 'react'
import CollectionRenameDialog from './collection-rename-dialog'

import { create, act } from 'react-test-renderer'

describe('CollectionRenameDialog', () => {
	let defaultProps

	beforeEach(() => {
		jest.resetAllMocks()

		defaultProps = {
			collection: {
				id: 'mockCollectionId',
				title: 'mockCollectionTitle'
			},
			onAccept: jest.fn(),
			onClose: jest.fn()
		}
	})

	test('renders with expected standard props', () => {
		defaultProps.collectionModules = null
		const component = create(<CollectionRenameDialog {...defaultProps} />)
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('input onChange event changes state and sets its own value properly', () => {
		defaultProps.collectionModules = null
		const reusableComponent = <CollectionRenameDialog {...defaultProps} />
		let component
		act(() => {
			component = create(reusableComponent)
		})

		expect(component.root.findByType('input').props.value).toBe('mockCollectionTitle')

		const mockEventObject = {
			target: {
				value: 'mockInputValue'
			}
		}

		act(() => {
			component.root.findByType('input').props.onChange(mockEventObject)
			component.update(reusableComponent)
		})

		expect(component.root.findByType('input').props.value).toBe('mockInputValue')
	})

	test('clicking the "Accept" button calls props.onAccept and props.onClose', () => {
		defaultProps.collectionModules = null
		const reusableComponent = <CollectionRenameDialog {...defaultProps} />
		let component
		act(() => {
			component = create(reusableComponent)
		})

		const mockEventObject = {
			target: {
				value: 'mockInputValue'
			}
		}
		act(() => {
			component.root.findByType('input').props.onChange(mockEventObject)
			component.update(reusableComponent)
		})

		expect(component.root.findByType('input').props.value).toBe('mockInputValue')

		act(() => {
			component.root.findByProps({ className: 'done-button secondary-button' }).props.onClick()
		})

		expect(defaultProps.onAccept).toHaveBeenCalledTimes(1)
		expect(defaultProps.onAccept).toHaveBeenCalledWith('mockCollectionId', 'mockInputValue')

		expect(defaultProps.onClose).toHaveBeenCalledTimes(1)
	})

	test('clicking the "x" button calls props.onClose', () => {
		defaultProps.collectionModules = null
		let component
		act(() => {
			component = create(<CollectionRenameDialog {...defaultProps} />)
		})

		act(() => {
			component.root.findByProps({ className: 'close-button' }).props.onClick()
		})

		expect(defaultProps.onClose).toHaveBeenCalledTimes(1)
	})
})
