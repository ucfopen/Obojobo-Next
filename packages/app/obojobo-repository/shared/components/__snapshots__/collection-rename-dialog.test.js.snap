// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionRenameDialog renders with expected standard props 1`] = `
<div
  className="collection-rename-dialog"
>
  <div
    className="top-bar"
  >
    <div
      className="collection-title"
    />
    <button
      aria-label="Close"
      className="repository--button close-button"
      onClick={[Function]}
    >
      Ã—
    </button>
  </div>
  <div
    className="wrapper"
  >
    <input
      aria-label="Rename Collection"
      className="collection-rename-input"
      onChange={[Function]}
      value="mockCollectionTitle"
    />
    <button
      className="repository--button done-button secondary-button"
      onClick={[Function]}
    >
      Accept
    </button>
  </div>
</div>
`;
